MekHQ/src/mekhq/campaign/mission/atb/scenario/AllyRescueBuiltInScenario.java

```diff
@@ -93,34 +97,38 @@ public void setExtraMissionForces(Campaign campaign, ArrayList<Entity> allyEntit
         setStart(Board.START_S);
         setDeploymentDelay(12);
 
+        final AtBContract contract = getContract(campaign);
+
         for (int i = 0; i < 4; i++) {
-            getAlliesPlayer().add(getEntity(getContract(campaign).getEmployerCode(),
-                    getContract(campaign).getAllySkill(), getContract(campaign).getAllyQuality(), UnitType.MEK,
-                    UnitMarket.getRandomWeight(UnitType.MEK, getContract(campaign).getEmployerCode(),
-                            campaign.getCampaignOptions().getRegionalMechVariations()),
+            getAlliesPlayer().add(getEntity(contract.getEmployerCode(), contract.getAllySkill(),
+                    contract.getAllyQuality(), UnitType.MEK,
+                    AtBMonthlyUnitMarket.getRandomWeight(campaign, UnitType.MEK, contract.getEmployerFaction()),
                     campaign));
         }
 
-        ArrayList<Entity> otherForce = new ArrayList<Entity>();
+        List<Entity> otherForce = new ArrayList<>();
 
         for (int i = 0; i < 8; i++) {
-            otherForce.add(getEntity(getContract(campaign).getEmployerCode(), getContract(campaign).getAllySkill(),
-                    getContract(campaign).getAllyQuality(), UnitType.MEK, UnitMarket.getRandomAeroWeight(), // max
-                                                                                                            // heavy
-                    campaign));
+            int weightClass;
+            do {
+                weightClass = AtBMonthlyUnitMarket.getRandomWeight(campaign, UnitType.MEK, contract.getEmployerFaction());
+            } while (weightClass == EntityWeightClass.WEIGHT_ASSAULT);
```

Comment:
Should we exclude superheavies too?
```suggestion
            } while (weightClass >= EntityWeightClass.WEIGHT_ASSAULT);
```

Reply:
Same as above